src/librustc_mir_build/hair/cx/expr.rs

```diff
@@ -753,10 +752,14 @@ fn convert_path_expr<'a, 'tcx>(
                     adt_def,
                     variant_index: adt_def.variant_index_with_ctor_id(def_id),
                     substs,
-                    user_ty: user_provided_type,
+                    user_ty,
                     fields: vec![],
                     base: None,
                 },
+                _ if ty.references_error() => {
+                    // Handle degenerate input without ICE (#67377).
+                    ExprKind::Literal { literal: ty::Const::zero_sized(cx.tcx, ty), user_ty: None }
```

Comment:
what if you use `cx.tcx.types.err` instead of `ty` here? That should avoid the `TooGeneric` and instead give us an `AlreadyReported`

Reply:
Pushing that change as it is reasonable, but there's no change in the output. Could there be a better placeholder than `ExprKind::Literal` or `ty::Const::zero_sized`?

Reply:
What do you mean by "change in the output"? If you change the `delay_span_bug` back to `span_bug`, will it still get triggered? I don't actually expect any stderr changes.

Reply:
ping @estebank did you have a chance to try this?

Reply:
Made the requested change, reverting `delay_span_bug` change and making `ty` `types.err`. The `span_bug` triggers.

Reply:
Thanks for trying! I guess at this point in parser recovery we just have to give up.